File: PaseoFrontEnd/src/app/modelos/productor.ts
import { Estado } from "../utils/estado";
import { Producto } from "./producto";
import { Image } from "./image";

export class Productor {
    id:number;
    nombre:string;
    descripcion:string;
    estado:Estado;
    productos:Producto[];
    imagenes:Image[];


	constructor( nombre: string, descripcion: string, estado: Estado, productos: Producto[], imagenes: Image[]) {
		this.nombre = nombre;
		this.descripcion = descripcion;
		this.estado = estado;
		this.productos = productos;
		this.imagenes = imagenes;
	}
    

    /**
     * Getter id
     * @return {number}
     */
	public get Id(): number {
		return this.id;
	}

    /**
     * Getter nombre
     * @return {string}
     */
	public get Nombre(): string {
		return this.nombre;
	}

    /**
     * Getter descripcion
     * @return {string}
     */
	public get Descripcion(): string {
		return this.descripcion;
	}

    /**
     * Getter estado
     * @return {Estado}
     */
	public get Estado(): Estado {
		return this.estado;
	}

    /**
     * Getter productos
     * @return {Producto[]}
     */
	public get Productos(): Producto[] {
		return this.productos;
	}

    /**
     * Getter imagenes
     * @return {Image[]}
     */
	public get Imagenes(): Image[] {
		return this.imagenes;
	}

    /**
     * Setter id
     * @param {number} value
     */
	public set Id(value: number) {
		this.id = value;
	}

    /**
     * Setter nombre
     * @param {string} value
     */
	public set Nombre(value: string) {
		this.nombre = value;
	}

    /**
     * Setter descripcion
     * @param {string} value
     */
	public set Descripcion(value: string) {
		this.descripcion = value;
	}

    /**
     * Setter estado
     * @param {Estado} value
     */
	public set Estado(value: Estado) {
		this.estado = value;
	}

    /**
     * Setter productos
     * @param {Producto[]} value
     */
	public set Productos(value: Producto[]) {
		this.productos = value;
	}

    /**
     * Setter imagenes
     * @param {Image[]} value
     */
	public set Imagenes(value: Image[]) {
		this.imagenes = value;
	}

}

File: Paseo2/src/main/java/modelos/Productor.java
package modelos;

import java.util.List;
import java.util.ArrayList;
import javax.persistence.*;

import com.fasterxml.jackson.annotation.JsonIgnore;

import utils.Estado;

@Entity
public class Productor {
	
	 @Id @GeneratedValue(strategy=GenerationType.IDENTITY) 
	 private Long id;
     private String nombre;
     private String descripcion;
     
     @Enumerated(EnumType.ORDINAL)
     private Estado estado;
     
     @OneToMany(mappedBy = "productor")
     @JsonIgnore
     private List<Producto> productos= new ArrayList<>();
     
     @OneToMany
     @JsonIgnore
     private List<Image> imagenes;
     
     public Productor() {
    	 
     }
     
	public Productor(String nombre, String descripcion, Estado estado, List<Producto> productos,
			List<Image> imagenes) {
		super();
		this.nombre = nombre;
		this.descripcion = descripcion;
		this.estado = estado;
		this.productos = productos;
		this.imagenes = imagenes;
	}
	
	public List<Image> getImagenes() {
		return imagenes;
	}
	public void setImagenes(List<Image> imagenes) {
		this.imagenes = imagenes;
	}
	public Long getId() {
		return id;
	}
	public void setId(Long id) {
		this.id = id;
	}
	public String getNombre() {
		return nombre;
	}
	public void setNombre(String nombre) {
		this.nombre = nombre;
	}
	public Estado getEstado() {
		return estado;
	}
	public void setEstado(Estado estado) {
		this.estado = estado;
	}
	public String getDescripcion() {
		return descripcion;
	}
	public void setDescripcion(String descripcion) {
		this.descripcion = descripcion;
	}
	public List<Producto> getProductos() {
		return productos;
	}
	public void setProductos(List<Producto> productos) {
		this.productos = productos;
	}
	@Override
	public String toString() {
		return "Productor [id=" + id + ", nombre=" + nombre + ", descripcion=" + descripcion + ", estado=" + estado + ", productos=" + productos + ", imagenes=" + imagenes + "]";
	}
	
	public String imprimirProductos() {
		return productos.toString();
	}
     
}
